dist/Storable/Storable.xs

```diff
@@ -5650,8 +5664,16 @@ static SV *get_lstring(pTHX_ stcxt_t *cxt, UV len, int isutf8, const char *cname
     stash = cname ? gv_stashpv(cname, GV_ADD) : 0;
     SEEN_NN(sv, stash, 0);	/* Associate this new scalar with tag "tagnum" */
 
-    if (len ==  0) {
+    /* 0 length strings should generally never be encoded in this 'long'
+     * format. this should only happen if artifically forced to do so, which
+     * only happens for booleans.
+     */
+    if (len == 0) {
+#ifdef sv_setbool
+        sv_setbool(sv, FALSE);
+#else
         SvPVCLEAR(sv);
+#endif
```

Comment:
get_lstring() is used for fetching all 3 string layouts, not just the long forms.

Reply:
Well that's a pretty silly mistake. I could certainly get this working, but I'm not sure if it's really worth it. This is rather weird hack to begin with.